docs/developers/keeper-bots/guides/use-borrowed-funds-for-cleaning.md

```diff
@@ -0,0 +1,56 @@
+---
+sidebar_position: 1
+---
+
+# Using borrowed funds for cleaning
+
+Mangrove's `cleanByImpersonation` function allows anyone to clean a failing offer, by providing the execution parameters that will make it fail. The function will execute the specified offers, including token transfers, and (1) reward the cleaner and remove the offer from the book if the offer fails or (2) revert the offer execution if it succeeds and keep the offer in the book; In both cases, all token transfers are reverted.
+
+Since this cleaning involves actual token transfers (though they will be reverted) Mangrove requires the caller to specify a taker account from which the inbound tokens will be transferred. In effect, this taker account will be impersonated during cleaning, hence `*ByImpersonation`.
+
+The taker must of course have approved Mangrove for transferring the inbound token on their behalf. And cleaners are free to specify _any_ account, including accounts they do not control.
+
+:::info Impersonated takers are unaffected
+The takers that are impersonated during cleaning are unaffected: All token transfers made inside `cleanByImpersonation` will be reverted.
+:::
+
+The prerequisites and steps needed to use the `Mangrove.cleanByImpersonation` function are:
+
+**Prerequisites**
+
+- `taker` must have approved Mangrove for spending `inbound_tkn` on the offer list you want to clean.
+- `taker` must have sufficient `inbound_tkn` funds for the execute arguments
+
+**Steps**
+
+1. Call `Mangrove.cleanByImpersonation(olKey, targets, taker)` specifying the offer list, an array of offers to clean, and the taker to impersonate.
+    - The targert are `MgvLib.CleanTarget` structs which has the following fields:
+        - `offerId`: The offer to clean
```

Comment:
Nitpick: End `offerId` bullet-point with a full stop (like the other points).

Reply:
Addressed in [e7881cd](https://github.com/mangrovedao/mangrove-docs/pull/204/commits/e7881cda67178bbc9ad90f64f55d9bd3fbda3c6a)